### ISSUE TITLE
Version 1.0.2 web build failed with Dart 3

### ISSUE THREAD

**Issue by timekone** [Zekfad/fetch_api#3]
I get the following error when building a flutter project for web:

```
.pub-cache/hosted/pub.dev/fetch_api-1.0.2/lib/src/iterator.dart:24:30:
Error: `@staticInterop` classes cannot have external extension members with type parameters.
  external IteratorResult<T> next();
                             ^
Error: Compilation failed.
```

Seems to be related to https://github.com/dart-lang/sdk/issues/49350

Flutter 3.13.6
Dart 3.1.3

**Comment by Zekfad** [comment]
Ah yes, I totally forgot why I made `Iterator` not a static interop in a first place.
Looks like this class will require another damn wrapper to get it working nicely.
This pesky `staticInterop` prohibits generics, because Dart runtime don't "own" the external object:
ref: https://github.com/dart-lang/sdk/issues/48327#issuecomment-1032810603

**Comment by Zekfad** [comment]
@timekone what version you're on, as I've tested code with `Flutter 3.16.0-6.0.pre.12 • channel master` and had no issues with both `ddc` and `dart2js` compilers?

**Comment by timekone** [comment]
@Zekfad I'm on `Channel stable, 3.13.6`. Using the default (`dart2js`) compiler.

**Comment by srujzs** [comment]
This is fixed in 3.2, you should be able to use generics in `@staticInterop` types. This is the related commit: https://github.com/dart-lang/sdk/commit/7d6414a6d5c0c1f7b419d589f6e84c4a15ce2ff5. Note that if you import `dart:js_interop`, there are some restrictions around what the type parameters can be bound to.
